src/builder.rs

```diff
@@ -356,9 +356,6 @@ impl DoubleArrayAhoCorasickBuilder {
             } else {
                 let fail_idx = state_id_map[fail_id as usize];
                 debug_assert_ne!(fail_idx, DEAD_STATE_IDX);
-                if fail_idx > FAIL_MAX {
-                    return Err(DaachorseError::automaton_scale("fail_idx", FAIL_MAX));
-                }
```

Comment:
Instead of removing the block, we should handle an exception if the number of patterns exceeds 2^24.
Could you add such a handling?

Reply:
updated in https://github.com/daac-tools/daachorse/pull/32/commits/c9a8ca8704b87de3ee3333ffed8a76c6536f5823.
Could you check the corner case since I don't have much confidence about it.

Reply:
Sure, and it's correct.